rclpy/rclpy/node.py

```diff
@@ -1150,9 +1151,14 @@ def create_publisher(
 
         publisher_handle = Handle(publisher_capsule)
 
+        if event_callbacks:
+            event_callbacks_with_logger = copy.copy(event_callbacks)
+        else:
+            event_callbacks_with_logger = PublisherEventCallbacks()
+        event_callbacks_with_logger._logger = self.get_logger()
```

Comment:
This is extremely hacky. If somebody creates a custom `PublisherEventCallbacks` with the default subscription and they doesn't add `_logger` member, then the callback will fail.

Consider either moving the callbacks definitions from `PublisherEventCallbacks`/`SubscriptionEventCallbacks` to `Publisher`/`Subscription`, and adding the default subscriptions there when `use_default_callbacks` option is true, or any other solution that avoids this workaround.

Reply:
I have taken a different approach. Please re-review the pull request.

Reply:
Thanks!